### ISSUE TITLE
Cannot connect to secure Windows 10 Workstation

### ISSUE THREAD

**Issue by jamielapointe** [FreeRDP/FreeRDP#6225]
Remote Computer: Windows 10 Enterprise LTSC Version 1809 OSBuild: 17763.1192
Host Computer: RHEL 7.8
freerdp version: xfreerdp 2.0.0-rc4 (n/a) - using the default version available from the RHEL 7.8 repo

The remote computer is secured according to the DISA STIG requirements.
Some of the remote computer RDP server settings
```
SecurityLayer = 2
MinEncryptionLevel = 3
AllowSecProtocolNegotiation = 1
PortNumber = 3389
SelectNetworkDetect = 1
SelectTransport = 2
UserAuthentication = 1
```
I am pretty sure it is setup to use NLA, FIPS, and possibly Kerebos

The two systems are on the same subnetwork with no gateways nor proxies in between.

I entered my actual domain, username, and machine name in the appropriate areas in the below commands.  I attempted variants of the domain such as:

`/u:DOMAIN\username`
`/d:DOMAIN /u:username`
`/u:username@DOMAIN.com`

Attempted unsuccessfully to connect to the remote computer from the host computer via xfreerdp and the following commands:

```
xfreerdp /log-level:TRACE /u:DOMAIN\username /v:machine
[DEBUG][com.freerdp.channels.cliprdr.client] - VirtualChannelEntryEx
[INFO][com.freerdp.client.common.cmdline] - loading channelEx cliprdr
[DEBUG][com.freerdp.client.x11] - Searching for XInput pointer device
[DEBUG][com.freerdp.client.x11] - Pointer device: 8
[DEBUG][com.freerdp.core.nego] - Enabling security layer negotiation: TRUE
[DEBUG][com.freerdp.core.nego] - Enabling restricted admin mode: FALSE
[DEBUG][com.freerdp.core.nego] - Enabling RDP security: TRUE
[DEBUG][com.freerdp.core.nego] - Enabling TLS security: TRUE
[DEBUG][com.freerdp.core.nego] - Enabling NLA security: FALSE
[DEBUG][com.freerdp.core.nego] - Enabling NLA extended security: FALSE
[DEBUG][com.freerdp.core.nego] - state: NEGO_STATE_TLS
[DEBUG][com.freerdp.core.nego] - Attempting TLS security
[DEBUG][com.freerdp.core] - connecting to peer 10.42.163.169
[DEBUG][com.freerdp.core.nego] - RequestedProtocols: 1
[DEBUG][com.freerdp.core.nego] - RDP_NEG_FAILURE
[WARN][com.freerdp.core.nego] - Error: HYBRID_REQUIRED_BY_SERVER
[DEBUG][com.freerdp.core.nego] - state: NEGO_STATE_RDP
[DEBUG][com.freerdp.core.nego] - Attempting RDP security
[DEBUG][com.freerdp.core] - connecting to peer 10.42.163.169
[DEBUG][com.freerdp.core.nego] - RequestedProtocols: 0
[ERROR][com.freerdp.core.transport] - BIO_read returned a system error 104: Connection reset by peer
[ERROR][com.freerdp.core.nego] - Protocol Security Negotiation Failure
[ERROR][com.freerdp.core] - freerdp_set_last_error ERRCONNECT_SECURITY_NEGO_CONNECT_FAILED [0x0002000C]
[ERROR][com.freerdp.core.connection] - Error: protocol security negotiation or connection failure
```

If I try to force NLA it selects nothing

```
[DEBUG][com.freerdp.channels.cliprdr.client] - VirtualChannelEntryEx
[INFO][com.freerdp.client.common.cmdline] - loading channelEx cliprdr
[DEBUG][com.freerdp.client.x11] - Searching for XInput pointer device
[DEBUG][com.freerdp.client.x11] - Pointer device: 8
[DEBUG][com.freerdp.core.nego] - Enabling security layer negotiation: TRUE
[DEBUG][com.freerdp.core.nego] - Enabling restricted admin mode: FALSE
[DEBUG][com.freerdp.core.nego] - Enabling RDP security: FALSE
[DEBUG][com.freerdp.core.nego] - Enabling TLS security: FALSE
[DEBUG][com.freerdp.core.nego] - Enabling NLA security: FALSE
[DEBUG][com.freerdp.core.nego] - Enabling NLA extended security: FALSE
[ERROR][com.freerdp.core.nego] - No security protocol is enabled
[ERROR][com.freerdp.core] - freerdp_set_last_error ERRCONNECT_SECURITY_NEGO_CONNECT_FAILED [0x0002000C]
[ERROR][com.freerdp.core.connection] - Error: protocol security negotiation or connection failure
```

If I try NLA extended, I get a different error:

```
[DEBUG][com.freerdp.channels.cliprdr.client] - VirtualChannelEntryEx
[INFO][com.freerdp.client.common.cmdline] - loading channelEx cliprdr
[DEBUG][com.freerdp.client.x11] - Searching for XInput pointer device
[DEBUG][com.freerdp.client.x11] - Pointer device: 8
[DEBUG][com.freerdp.core.nego] - Enabling security layer negotiation: TRUE
[DEBUG][com.freerdp.core.nego] - Enabling restricted admin mode: FALSE
[DEBUG][com.freerdp.core.nego] - Enabling RDP security: TRUE
[DEBUG][com.freerdp.core.nego] - Enabling TLS security: TRUE
[DEBUG][com.freerdp.core.nego] - Enabling NLA security: FALSE
[DEBUG][com.freerdp.core.nego] - Enabling NLA extended security: TRUE
[DEBUG][com.freerdp.core.nego] - state: NEGO_STATE_EXT
[DEBUG][com.freerdp.core.nego] - Attempting NLA extended security
[DEBUG][com.freerdp.core] - connecting to peer 10.42.163.169
[DEBUG][com.freerdp.core.nego] - RequestedProtocols: 11
[DEBUG][com.freerdp.core.nego] - RDP_NEG_RSP
[DEBUG][com.freerdp.core.nego] - selected_protocol: 8
[DEBUG][com.freerdp.core.nego] - state: NEGO_STATE_FINAL
[DEBUG][com.freerdp.core.nego] - Negotiated EXT security
[ERROR][com.freerdp.core.nego] - cannot connect security layer because no protocol has been selected yet.
[DEBUG][com.freerdp.core.nego] - Failed to connect with EXT security
[ERROR][com.freerdp.core] - freerdp_set_last_error ERRCONNECT_SECURITY_NEGO_CONNECT_FAILED [0x0002000C]
[ERROR][com.freerdp.core.connection] - Error: protocol security negotiation or connection failure
```

I tried enabling FIPS via xfreerdp, but this had no effect.
This _may_ have something to do with Kerebos... I am not exactly sure.  I can ask IT for more information on how the Windows 10 workstation is setup exactly.

I can connect to the secure Windows 10 Workstation from another Windows 10 workstation using the standard Microsoft Remote Desktop Connection client.  I tried saving off the desktop.rdp and having xfreerdp call it to no avail

Please let me know if there is something else I should try to help debug and/or resolve this issue or if I should try with the latest release of freeRDP.

**Comment by ondrejholy** [comment]
Currently, it is not possible to connect to such server, because NLA security is based on NTLM which is not FIPS compliant. There is an experimental Kerberos implementation for FreeRDP, but it is disabled on RHEL and other distributions, because of the big amount of pending bugs...

**Comment by zeclab** [comment]
Hi,
Is there news as to when this might be resolved?
Struggling with users who are in the "Protected Users" group in Active Directory and suspect that this is related.
Thanks

**Comment by akallabeth** [comment]
@zeclab https://github.com/FreeRDP/FreeRDP/issues/5258

**Comment by akallabeth** [comment]
kerberos is implemented and working with the 3.0 branches, so this issue is resolved (from our point of view)
